### ISSUE TITLE
Modal does not trap focus if shift+tab is immediately pressed

### ISSUE THREAD

**Issue by d-lane** [angular-ui/bootstrap#5229]
By opening a ui.bootstrap.modal which does not have an initially focused element and then pressing shift+tab the keyboard focus will not be trapped within the modal. For instance, if you press the button entitled 'Open me!' on the [demo page](https://angular-ui.github.io/bootstrap/) and press shift+tab a few times the keyboard focus will be behind the dialog.

**Comment by wesleycho** [comment]
This seems strange, as we have code specifically for this situation, as well as a unit test against this.

**Comment by wesleycho** [comment]
What is happening here is that the modal does not have focus when it is opened.

The question is, should the modal have focus? On research, the answer appears to be yes. PRs are welcome for moving the focus to the modal window element on instantiation - a simple unit test checking for focus is necessary. This also should not interfere with the `autofocus` attribute.
